server/graphql/common/expenses.ts

```diff
@@ -1599,16 +1616,33 @@ const editOnlyTagsAndAccountingCategory = async (
       throw new Unauthorized("You don't have permission to edit the accounting category for this expense");
     }
 
-    newValues['AccountingCategoryId'] = expenseData.accountingCategory?.id || null;
+    const userRole = getUserRole(req.remoteUser, expense.collective);
+    updateClauses.push(`"AccountingCategoryId" = :AccountingCategoryId`);
+    updateClauses.push(
+      `"data" = ${deepJSONBSet('data', ['valuesByRole', userRole, 'accountingCategory'], ':accountingCategory')}`,
+    );
   }
 
-  if (isEmpty(newValues)) {
+  if (isEmpty(updateClauses)) {
     return expense;
   }
 
-  const updatedExpense = await expense.update(newValues);
-  updatedExpense.createActivity(activities.COLLECTIVE_EXPENSE_UPDATED, req.remoteUser);
-  return updatedExpense;
+  // Use a raw query to unlock the ability of using `JSONB_SET`, which will prevent concurrency issues on the `data` field
```

Comment:
This was important since updating tags and accounting categories can be done regardless of the status, which means it could be edited at the same time it is being processed.

I also tried with a `sequelize.literal`, but `replacements` is not supported for updates, so it was unclear how to pass sanitized values.

Reply:
Cool, I understand why and agree this is a good idea here.